### ISSUE TITLE
[hunt] sklearn_transformations.py:316 — With params={} (the evidenced workload), self.params_num =…

### ISSUE THREAD

**Issue by anpilove** [anpilove/FEDOT#14]
Кандидат найден EvolveAgent'ом (LLM‑hunt по исходникам FEDOT, модель `z-ai/glm-5.3-flash`), прошёл технический экран и поставлен в очередь на часовую оценку `Fedot(preset="best_quality", timeout=3600)`. Это **гипотеза агента**, не подтверждённый дефект: вердикт по качеству ниже.

## Где

- Файл: [`fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py`](https://github.com/aimclub/FEDOT/blob/8990f28b0664c99f7eb2fddcee3931151fe16e79/fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py#L316) строка 316
- Тип гипотезы: улучшение качества
- FEDOT commit: `8990f28` (0.7.5)
- Кандидат `021e28ae27a7`, run `20260916-043211-3f811fa2`, 2026-09-16

## Гипотеза агента

With params={} (the evidenced workload), self.params_num = self.params.to_dict() carries no strategy, so the numerical imputer uses sklearn's default strategy='mean'. Mean imputation is computed from training moments that are strongly distorted when missingness co-occurs with skewed or heavy-tailed distributions (a common reason values became NaN after replace_inf_with_nans on line 373). Each imputed cell is then set to the mean, shifting the feature distribution, and after the downstream scaling operation in the evidenced pipeline (simple_imputation -> scaling -> logit) every imputed row lands exactly at the feature mean in standardized space, biasing the fitted logistic coefficients on features with many missing entries. Median imputation, computed from the same fitted state without new dependencies, is robust to this distortion and changes the imputed values only where NaNs exist.

**Предлагаемое изменение.** In __init__ (line 316), give the numerical imputer a robust default while preserving explicit user choices: self.params_num = {'strategy': 'median', **self.params.to_dict()}. The categorical imputer (strategy='most_frequent') and all fit/transform logic, column division, binary-feature handling and merging stay byte-identical; only the fill value used for numerical NaNs changes when the user supplied no strategy.

**Ожидаемый эффект.** For tabular pipelines containing simple_imputation with missing numerical values, fitted downstream models see imputed rows closer to the distributional center of each feature under skew/outliers, changing standardized inputs to logit/lasso/ridge-style models. Metrics such as accuracy/ROC-AUC (classification) or RMSE/MAE (regression) may improve where mean fill currently biases imputed rows; they cannot change on datasets without NaNs, and pipelines without simple_imputation are untouched.

## Патч

```diff
--- a/fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py
+++ b/fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py
@@
-        self.params_num = self.params.to_dict()
+        self.params_num = {'strategy': 'median', **self.params.to_dict()}
```

## Технический экран hunt

- Behavior probe (stock vs patched): `changed`; mechanism_supported_pending_independent_gates
- Comparative pytest FEDOT: `test_failures`
- Приоритет в очереди: `normal` (hunt_enqueue_no_toy_veto)

## Часовой Fedot (judge)

| Задача | Результат | δ ROC‑AUC/MAE |
|---|---|---|
| `openml-10101-blood-transfusion` | regression openml-10101-blood-transfusion delta -0.0480 | -0.0480 |

Stock ROC‑AUC на `openml-10101-blood-transfusion` = 0.694, patched = 0.646, порог min_delta = 0.01.

---
Источник: [FEDOT.LLM / EvolveAgent](https://github.com/anpilove/FEDOT.LLM/tree/research/evolve-agent-cloud/fedotllm/agents/evolve). Stock‑кэш judge считался до исправления чтения OpenML split (0‑based rowid), вердикты подлежат пересчёту.
